From be13829bb4ac967e41da4bde8c3c175c1b4b73dc Mon Sep 17 00:00:00 2001
From: Dung Cao <dung@os.amperecomputing.com>
Date: Thu, 11 Jul 2024 07:50:35 +0000
Subject: [PATCH 5/5] Prevent too many dump is captured simultaneously

Not only prevent too many dump is captured simultaneously for
dumpType=UserRequested but also need to prevent for all of dump types.

Signed-off-by: Dung Cao <dung@os.amperecomputing.com>
---
 dump_manager_bmc.cpp | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/dump_manager_bmc.cpp b/dump_manager_bmc.cpp
index 1fa0acd..7fe7e9d 100644
--- a/dump_manager_bmc.cpp
+++ b/dump_manager_bmc.cpp
@@ -63,7 +63,7 @@ sdbusplus::message::object_path
     std::string path = extractParameter<std::string>(
         convertCreateParametersToString(CreateParameters::FilePath), params);
 
-    if ((Manager::fUserDumpInProgress == true) && (dumpType == DumpTypes::USER))
+    if (Manager::fUserDumpInProgress == true)
     {
         lg2::info("Another user initiated dump in progress");
         elog<sdbusplus::xyz::openbmc_project::Common::Error::Unavailable>();
@@ -98,10 +98,8 @@ sdbusplus::message::object_path
         elog<InternalFailure>();
     }
 
-    if (dumpType == DumpTypes::USER)
-    {
-        Manager::fUserDumpInProgress = true;
-    }
+    Manager::fUserDumpInProgress = true;
+
     return objPath.string();
 }
 
@@ -133,11 +131,8 @@ uint32_t Manager::captureDump(DumpTypes type, const std::string& path)
     else if (pid > 0)
     {
         Child::Callback callback = [this, type, pid](Child&, const siginfo_t*) {
-            if (type == DumpTypes::USER)
-            {
-                lg2::info("User initiated dump completed, resetting flag");
-                Manager::fUserDumpInProgress = false;
-            }
+            lg2::info("User initiated dump completed, resetting flag");
+            Manager::fUserDumpInProgress = false;
             this->childPtrMap.erase(pid);
         };
         try
-- 
2.34.1

